include/deal.II/base/numbers.h

```diff
@@ -318,7 +318,7 @@ namespace numbers
   typename NumberTraits<number>::real_type
   NumberTraits<number>::abs (const number &x)
   {
-    return std::fabs(x);
+    return std::fabs((double)x);
```

Comment:
This change has to be disussed. How to do this properly? MSVC seems to need an explicit cast here.